tests/test_parameter_tiv.py

```diff
@@ -0,0 +1,400 @@
+from eitprocessing.datahandling.intervaldata import IntervalData
+from eitprocessing.datahandling.sequence import Sequence
+from eitprocessing.parameters.tidal_impedance_variation import TIV
+
+environment = Path(
+    os.environ.get(
+        "EIT_PROCESSING_TEST_DATA",
+        Path(__file__).parent.parent.resolve(),
+    ),
+)
+data_directory = environment / "tests" / "test_data"
+draeger_file1 = data_directory / "Draeger_Test3.bin"
+timpel_file = data_directory / "Timpel_Test.txt"
+
+
+class MockEITData(EITData):
+    """Class to create Mock EITData for running tests."""
+
+    def __init__(self):
+        """Mock pixel_impedance.
+
+        Each pixel is an identical sine wave with a baseline drop to create difference between
+        inspiratory and expiratory limb for one of the breaths.
+        """
+        duration = 10
+        fs = 1000
+        frequency = 0.25
+        amplitude = 1.0
+        baseline_drop = -0.5
+        # Calculate the period of the sine wave
+        period = 1 / frequency
+
+        # Create time vector with extra breaths
+        extra_breaths_duration = 2 * period  # One period for the extra breath at start and end
+        total_duration = duration + extra_breaths_duration
+        t = np.linspace(0, total_duration, int(total_duration * fs), endpoint=False)
+
+        # Generate the sine wave
+        signal = amplitude * np.sin(2 * np.pi * frequency * t)
+
+        # Apply baseline drop halfway through the original duration
+        halfway_index = len(signal) // 2
+        signal[halfway_index:] += baseline_drop  # Shift the baseline down for the second half
+
+        # Store the waves in a (time, 2, 2) array
+        pixel_impedance = np.empty((len(t), 2, 2))
+        pixel_impedance[:, 0, 0] = signal
+        pixel_impedance[:, 0, 1] = signal
+        pixel_impedance[:, 1, 0] = signal
+        pixel_impedance[:, 1, 1] = signal
+
+        self.time = t
+        self.pixel_impedance = pixel_impedance
+        self.label = "raw"
+
+
+class MockContinuousData(ContinuousData):
+    """Class to create Mock ContinuousData for running tests."""
+
+    def __init__(self, mock_eit_data: MockEITData):
+        pixel_impedance = mock_eit_data.pixel_impedance
+        self.time = mock_eit_data.time
+        self.values = np.nansum(
+            pixel_impedance,
+            axis=(1, 2),
+        )
+        self.label = "global_impedance(raw)"
+        self.sample_frequency = 1000
+
+
+class MockIntervalData:
+    """Class to create Mock IntervalData for running tests."""
+
+    def __init__(self):
+        self.data = []
+
+    def add(self, item: list):
+        """Function to add item to IntervalData."""
+        self.data.append(item)
+
+
+class MockSequence(Sequence):
+    """Class to create Mock Sequence for running tests."""
+
+    def __init__(self, mock_eit_data: MockEITData, mock_continuous_data: MockContinuousData):
+        self.eit_data = mock_eit_data
```

Comment:
If I change it to a data collection I need to initialise all the things that are needed for EITData but in the way I wrote it it works for the functions I want to test.

Reply:
It may work, but it is not clear nor correct. The attributes of the mocked sequence are not the attributes of normal sequences. That does not convince me the tests represent the actual code.